perceval/backends/remo.py

```diff
@@ -0,0 +1,275 @@
+        origin = origin if origin else url
+
+        super().__init__(origin, cache=cache)
+        self.url = url
+        self.client = ReMoClient(url)
+
+    @metadata
+    def fetch(self):
+        """Fetch events from the ReMo url.
+
+        The method retrieves, from a ReMo url, the
+        events.
+
+        :returns: a generator of events
+        """
+
+        logger.info("Looking for events at url '%s'", self.url)
+
+        self._purge_cache_queue()
+
+        nevents = 0  # number of events processed
+        tevents = 0  # number of events from API data
+        more_events = True # There are more events to be processed
+        events_next_uri = None # URI for the next events query
+
+        while more_events:
+            raw_events = self.client.get_events(events_next_uri)
+            self._push_cache_queue(raw_events)
+            events_data = json.loads(raw_events)
+
+            try:
+                logger.debug("Reps: %i/%i", nevents, tevents)
+
+                tevents = events_data['meta']['total_count']
+                events_next_uri = events_data['meta']['next']
+
+                if not events_next_uri:
+                    more_events = False
+
+                events = events_data['objects']
+            except:
+                cause = ("Bad JSON format for mozilla_events: %s" % (events_data))
+                raise ParseError(cause=cause)
+
+            for rep in events:
+                yield rep
+                nevents += 1
+
+            self._flush_cache_queue()
+
+        logger.info("Total number of events: %i (%i expected)", nevents, tevents)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the events from the cache.
+
+        :returns: a generator of events
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached bugs: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        nevents = 0
+
+        for items in cache_items:
+            events = json.loads(items)['objects']
+            for event in events:
+                nevents += 1
+                yield event
+
+        logger.info("Retrieval process completed: %s bugs retrieved from cache",
+                    nevents)
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+        return str(item['event_url'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a ReMo item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return float(parser.parse(item['end']).timestamp())
```

Comment:
Yes, and all of them use this parser.parse as basics. But yes, I can add convert to that if you are more comfortable with it. Done!